datafusion/optimizer/src/optimizer.rs

```diff
@@ -92,9 +92,12 @@ pub struct OptimizerContext {
 impl OptimizerContext {
     /// Create optimizer config
     pub fn new() -> Self {
+        let mut options = ConfigOptions::default();
+        options.optimizer.filter_null_join_keys = true;
```

Comment:
I double-checked this wasn't introduced by the recent config rework, and confirmed that it has always been the case that OptimizerContext defaults this to true, but ConfigOptions defaults it to false.

Reply:
My intention was always to have this default to false so this looks like a bug

Reply:
`OptimizerContext` is now only used within tests, and so I am going to leave this for now. If you feel strongly I can file a follow up PR to change this, but it creates a non-trivial amount of test-churn